lib/pbench/test/unit/server/test_cache_manager.py

```diff
@@ -969,36 +1006,61 @@ def test_cache_map_get_info_cmap(
             assert file_info == expected_msg
 
     @pytest.mark.parametrize(
-        "file_path, exp_file_type, exp_stream",
+        "file_path,is_unpacked,exp_stream",
         [
-            ("", CacheType.FILE, io.BytesIO(b"tarball_as_a_byte_stream")),
-            (None, CacheType.FILE, io.BytesIO(b"tarball_as_a_byte_stream")),
-            ("f1.json", CacheType.FILE, io.BytesIO(b"file_as_a_byte_stream")),
-            ("subdir1/f12_sym", CacheType.FILE, io.BytesIO(b"file1_as_a_byte_stream")),
+            ("", True, b"tarball_as_a_byte_stream"),
+            (None, True, b"tarball_as_a_byte_stream"),
+            ("f1.json", True, b"{'json': 'value'}"),
+            ("subdir1/f12_sym", False, CacheExtractBadPath(Path("a"), "b")),
         ],
     )
     def test_get_inventory(
-        self, make_logger, monkeypatch, tmp_path, file_path, exp_file_type, exp_stream
+        self, make_logger, monkeypatch, tmp_path, file_path, is_unpacked, exp_stream
     ):
-        """Test to extract file contents/stream from a file"""
-        tar = Path("/mock/dir_name.tar.xz")
-        cache = Path("/mock/.cache")
+        """Test to extract file contents/stream from a file
+
+        NOTE: we check explicitly whether Tarball.stream() chooses to unpack
+        the tarball, based on "is_unpacked", to be sure both cases are covered;
+        but be aware that accessing the tarball itself (the "" and None
+        file_path cases) don't ever unpack and should always have is_unpacked
+        set to True. (That is, we don't expect get_results to be called in
+        those cases.)
+        """
+        archive = tmp_path / "mock" / "archive" / "ABC"
+        archive.mkdir(parents=True, exist_ok=True)
+        tar = archive / "dir_name.tar.xz"
+        tar.write_bytes(b"tarball_as_a_byte_stream")
+        cache = tmp_path / "mock" / ".cache"
+        unpacked = cache / "ABC" / "dir_name"
+
+        get_results_called = False
+
+        def fake_results(self, lock: LockRef) -> Path:
+            nonlocal get_results_called
+            get_results_called = True
+            self.unpacked = unpacked
+            return self.unpacked
```

Comment:
I don't see any assertion checking `get_results_called`.  Since we go to the trouble of setting it, it seems like we should be checking it....  🙂

Reply:
That was a holdover from previous logic. Now `get_results` is *always* called, and decides internally (under a lock) whether to unpack. Since we're mocking that, and explicitly setting `tb.unpacked`, there's no point to a test (or in maintaining this variable).